### PR TITLE
Modal header left section

### PR DESCRIPTION
**Pull request by prasook-jain**
`title` prop is changed to `leftSection` to generalise use-case of left side of header, as `rightSection` prop
 have, but if `leftSection` is string, heading styles are added as default.

### CODE REVIEW

File: src/components/Modal/components/Header.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { IStyled, fontSizeType } from "../../../common/types"
  * the places instead of having them duplicated.
  */
 interface HeaderProps {
-  title: string
+  leftSection: ReactNode
```
Comment (Kushagra8888) [inline src/components/Modal/components/Header.tsx]: Shouldn't this be `ReactNode | string`

File: src/components/Modal/types.ts [hunk 2]
```diff
@@ -4,7 +4,13 @@ import { fontSizeType } from "../../common/types"
 
 export interface ModalWrapperProps extends BaseComponentProps {
   /** Contents to be rendered in the header section */
-  header: { title: string; fontSize?: fontSizeType, rightSection?: ReactNode, noFill?: boolean }
+  header: {
```
Comment (Kushagra8888) [inline src/components/Modal/types.ts]: When the `leftSection` is not a string, then the other props become irrelevant, so I'm thinking we can let `leftSection` be either a `ReactNode` or an object containing the required properties (`title`, `fontSize` etc)

File: src/components/Modal/components/Header.tsx (hunk 1 above)
Comment (prasook-jain) [reply src/components/Modal/components/Header.tsx]: ReactNode has string itself

Comment (Kushagra8888) [reply src/components/Modal/components/Header.tsx]: oh, okay

File: src/components/Modal/types.ts (hunk 2 above)
Comment (prasook-jain) [reply src/components/Modal/types.ts]: this seems better API changes pushed in latest commit, @Kushagra8888

**Pull request merged**